### PR TITLE
feat: Updates the package to support v2 of the REST API

### PR DESCRIPTION
**Pull request by scottlovegrove**

### CODE REVIEW

File: src/testUtils/testDefaults.ts [hunk 1]
```diff
@@ -37,30 +39,30 @@ export const DEFAULT_REQUEST_ID = 'ARequestID'
...
 export const INVALID_DUE_DATE = {
     ...DEFAULT_DUE_DATE,
-    recurring: 'false',
+    isRecurring: 'false',
 }
 
 export const DEFAULT_QUICK_ADD_RESPONSE: QuickAddTaskResponse = {
-    id: DEFAULT_TASK_ID,
-    projectId: DEFAULT_PROJECT_ID,
+    id: parseInt(DEFAULT_TASK_ID),
```
Comment (lefcha) [inline src/testUtils/testDefaults.ts:53]: Why do we convert these properties?

https://developer.todoist.com/sync/v9/#quick-add-an-item

File: src/testUtils/testDefaults.ts [hunk 2]
```diff
@@ -37,30 +39,30 @@ export const DEFAULT_REQUEST_ID = 'ARequestID'
...
     description: DEFAULT_TASK_DESCRIPTION,
     priority: DEFAULT_TASK_PRIORITY,
-    sectionId: DEFAULT_SECTION_ID,
-    parentId: DEFAULT_PARENT_ID,
+    sectionId: parseInt(DEFAULT_SECTION_ID),
+    parentId: parseInt(DEFAULT_PARENT_ID),
     childOrder: DEFAULT_ORDER,
-    labels: DEFAULT_LABELS,
-    responsibleUid: DEFAULT_ASSIGNEE,
+    labels: [1, 2, 3],
```
Comment (lefcha) [inline src/testUtils/testDefaults.ts:61]: Shouldn't these be strings?

https://developer.todoist.com/sync/v9/#quick-add-an-item

File: src/testUtils/testDefaults.ts [hunk 3]
```diff
@@ -37,30 +39,30 @@ export const DEFAULT_REQUEST_ID = 'ARequestID'
...
+    responsibleUid: parseInt(DEFAULT_ASSIGNEE),
     checked: 0,
     dateAdded: DEFAULT_DATE,
```
Comment (lefcha) [inline src/testUtils/testDefaults.ts:64]: Should be `addedAt`?

https://developer.todoist.com/sync/v9/#quick-add-an-item

File: src/testUtils/testDefaults.ts (hunk 1 above)
Comment (scottlovegrove) [reply src/testUtils/testDefaults.ts:53]: Ah, I was basing this still on v8 🤦🏻‍♂️ I'll update this PR to reflect the v9 changes.

File: src/testUtils/testDefaults.ts [hunk 4]
```diff
@@ -37,30 +39,30 @@ export const DEFAULT_REQUEST_ID = 'ARequestID'
...
     date: DEFAULT_DATE,
 }
 
 export const INVALID_DUE_DATE = {
     ...DEFAULT_DUE_DATE,
-    recurring: 'false',
+    isRecurring: 'false',
 }
 
 export const DEFAULT_QUICK_ADD_RESPONSE: QuickAddTaskResponse = {
```
Comment (lefcha) [inline src/testUtils/testDefaults.ts:52]: In general these responses should be updated to v9, see also some relevant comments below.

**Review by lefcha — APPROVED** [review]
Looks good overall, apart from some comments.

**Comment by scottlovegrove** [comment]
Great spot that I was still using v8 sync types @lefcha, have updated this now to use v9 and its types.

**Comment by lefcha** [comment]
Looks good now @scottlovegrove

**Pull request merged**